### ISSUE TITLE
[gatsby-image] only getting 1 and 1.5x w/ contentful

### ISSUE THREAD

**Issue by sarahbethfederman** [gatsbyjs/gatsby#10987]
## Summary
I'm wondering how to get a 2x image to work with gatsby-image and gatsby-source-contentful. I'm trying to render an image at 344 pixels wide with a srcset that includes a resolution of 2x (so 688px wide)

Here's my query:
```
          fixed(width: 344, quality: 100) {
            ...GatsbyContentfulFixed
          }
```

## Relevant information

Here's what is output: 
```
<picture>
<source srcset="//images.ctfassets.net/itcc7vu35nc1/42oobVWF3yYsUmaAs0yUyq/d807e1528f8f595ced9fc822dc44b77f/dropdown_2x.png?w=344&amp;h=352&amp;q=100 1x,
//images.ctfassets.net/itcc7vu35nc1/42oobVWF3yYsUmaAs0yUyq/d807e1528f8f595ced9fc822dc44b77f/dropdown_2x.png?w=516&amp;h=528&amp;q=100 1.5x">
<img alt="" width="344" height="352" src="//images.ctfassets.net/itcc7vu35nc1/42oobVWF3yYsUmaAs0yUyq/d807e1528f8f595ced9fc822dc44b77f/dropdown_2x.png?w=344&amp;q=100" style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center center; opacity: 1; transition: opacity 0.5s ease 0s;">
</picture>
```

Here's what I want to happen, and it works when I add the additional url and 2x in dev tools:
```
<picture>
<source srcset="//images.ctfassets.net/itcc7vu35nc1/42oobVWF3yYsUmaAs0yUyq/d807e1528f8f595ced9fc822dc44b77f/dropdown_2x.png?w=344&amp;h=352&amp;q=100 1x,
//images.ctfassets.net/itcc7vu35nc1/42oobVWF3yYsUmaAs0yUyq/d807e1528f8f595ced9fc822dc44b77f/dropdown_2x.png?w=516&amp;h=528&amp;q=100 1.5x, //images.ctfassets.net/itcc7vu35nc1/42oobVWF3yYsUmaAs0yUyq/d807e1528f8f595ced9fc822dc44b77f/dropdown_2x.png?w=688&amp;q=100 2x"><img alt="" width="344" height="352" src="//images.ctfassets.net/itcc7vu35nc1/42oobVWF3yYsUmaAs0yUyq/d807e1528f8f595ced9fc822dc44b77f/dropdown_2x.png?w=344&amp;q=100" style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center center; opacity: 1; transition: none 0s ease 0s;"></picture><noscript><picture><source srcSet="//images.ctfassets.net/itcc7vu35nc1/42oobVWF3yYsUmaAs0yUyq/d807e1528f8f595ced9fc822dc44b77f/dropdown_2x.png?w=344&h=352&q=100 1x,
//images.ctfassets.net/itcc7vu35nc1/42oobVWF3yYsUmaAs0yUyq/d807e1528f8f595ced9fc822dc44b77f/dropdown_2x.png?w=516&h=528&q=100 1.5x" />
<img width="344" height="352" src="//images.ctfassets.net/itcc7vu35nc1/42oobVWF3yYsUmaAs0yUyq/d807e1528f8f595ced9fc822dc44b77f/dropdown_2x.png?w=344&q=100" alt="" style="position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/>
</picture>
```

### Environment (if relevant)

```
  System:
    OS: macOS 10.14
    CPU: (8) x64 Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 8.9.4 - ~/.nvm/versions/node/v8.9.4/bin/node
    Yarn: 1.9.4 - /usr/local/bin/yarn
    npm: 6.5.0 - ~/.nvm/versions/node/v8.9.4/bin/npm
  Languages:
    Python: 3.6.5 - /Users/federman/anaconda3/bin/python
  Browsers:
    Chrome: 71.0.3578.98
    Firefox: 57.0.4
    Safari: 12.0
  npmPackages:
    gatsby: ^2.0.83 => 2.0.83 
    gatsby-image: ^2.0.25 => 2.0.25 
    gatsby-mdx: ^0.2.0 => 0.2.0 
    gatsby-plugin-react-helmet: ^3.0.5 => 3.0.5 
    gatsby-plugin-sass: ^2.0.7 => 2.0.7 
    gatsby-plugin-sharp: ^2.0.17 => 2.0.17 
    gatsby-plugin-styled-components: ^3.0.4 => 3.0.4 
    gatsby-source-contentful: ^2.0.22 => 2.0.22 
    gatsby-transformer-sharp: ^2.1.10 => 2.1.10 
  npmGlobalPackages:
    gatsby-cli: 2.4.8
    gatsby: 2.0.85
```

### File contents (if changed)

`gatsby-config.js`: 
```
require("dotenv").config();

module.exports = {
  siteMetadata: {
    title: "Spectrum Website"
  },
  plugins: [
    `gatsby-transformer-sharp`, // for gatsby-image
    `gatsby-plugin-sharp`,
    `gatsby-plugin-react-helmet`,
    `gatsby-plugin-styled-components`,
    {
      resolve: `gatsby-mdx`, // for making markdown into html
      options: {
        extensions: ['.md', '.mdx']
      },
    },
    '@contentful/gatsby-transformer-contentful-richtext',
    {
      resolve: `gatsby-plugin-sass`,
      options: {
        includePaths: ["node_modules/@spectrum/spectrum-css"],
      },
    },
    {
      resolve: `gatsby-source-contentful`,
      options: {
        spaceId: process.env.CONTENTFUL_SPACE_ID,
        accessToken: process.env.CONTENTFUL_ACCESS_TOKEN
      }
    }
  ]
};

```
`package.json`: 
```
{
  "name": "spectrum-site-platform",
  "description": "Spectrum Documentation Website",
  "version": "0.1.1",
  "author": "Sarah Federman <federman@adobe.com>",
  "dependencies": {
    "@contentful/gatsby-transformer-contentful-richtext": "^12.1.2",
    "@contentful/rich-text-html-renderer": "^10.3.0",
    "@contentful/rich-text-types": "^12.1.2",
    "@mdx-js/mdx": "^0.15.7",
    "@mdx-js/tag": "^0.15.6",
    "@react/react-spectrum": "^2.16.1",
    "@spectrum/spectrum-css": "^2.7.2",
    "@spectrum/spectrum-flexbox-grid": "^1.1.1",
    "babel-plugin-styled-components": "^1.10.0",
    "contentful": "^7.0.6",
    "core-js": "^2.6.1",
    "deepmerge": "^3.0.0",
    "dotenv": "^6.2.0",
    "escape-html": "^1.0.3",
    "gatsby": "^2.0.83",
    "gatsby-image": "^2.0.25",
    "gatsby-mdx": "^0.2.0",
    "gatsby-plugin-react-helmet": "^3.0.5",
    "gatsby-plugin-sass": "^2.0.7",
    "gatsby-plugin-sharp": "^2.0.17",
    "gatsby-plugin-styled-components": "^3.0.4",
    "gatsby-source-contentful": "^2.0.22",
    "gatsby-transformer-sharp": "^2.1.10",
    "node-sass": "^4.11.0",
    "npm-run-parallel": "^0.6.0",
    "nuka-carousel": "^4.4.5",
    "prop-types": "^15.6.2",
    "react": "^16.7.0",
    "react-dom": "^16.7.0",
    "react-helmet": "^5.2.0",
    "slugify": "^1.3.4",
    "styled-components": "^4.1.3"
  },
  "keywords": [
    "gatsby"
  ],
  "scripts": {
    "build": "npx gatsby build",
    "build:watch": "nodemon --watch src --exec 'npm run deploy'",
    "serve": "npx gatsby serve",
    "deploy": "npm run build && npm run serve",
    "develop": "npx gatsby develop",
    "format": "prettier --write './src/**/*.js'",
    "lint": "eslint './src/**/*.js'",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "lint-staged": {
    "src/**/*.{js,jsx}": [
      "prettier --write",
      "git add",
      "eslint"
    ]
  },
  "husky": {
    "hooks": {
      "pre-commit": "lint-staged"
    }
  },
  "devDependencies": {
    "babel-eslint": "^9.0.0",
    "eslint": "^5.11.1",
    "eslint-config-prettier": "^3.3.0",
    "eslint-config-react-app": "^3.0.6",
    "eslint-plugin-babel": "^5.3.0",
    "eslint-plugin-import": "^2.14.0",
    "eslint-plugin-jsx-a11y": "^6.1.2",
    "eslint-plugin-prettier": "^3.0.1",
    "husky": "^1.3.1",
    "lint-staged": "^8.1.0",
    "nodemon": "^1.18.9",
    "prettier": "^1.15.3"
  },
  "repository": {
    "type": "git",
    "url": "https://git.corp.adobe.com/Spectrum/spectrum-site-platform"
  }
}

```

**Comment by sidharthachatterjee** [comment]
So sorry you're seeing this @sarahbethfederman 

Just tried to reproduce this at https://github.com/sidharthachatterjee/contentful-playground but I can't seem to 

The live site is at https://silly-rosalind-090e60.netlify.com/

It seems to have 1x, 1.5x and 2x

```
<picture><source srcset="//images.ctfassets.net/9w4kzu6fppqz/2j9nxa9Ppi6sYUsUgSa2yy/90f8c2342c9578aafda73df2ffec24c1/thebiglebowskifeat-770x433.jpg?w=344&amp;h=193&amp;q=100 1x,
//images.ctfassets.net/9w4kzu6fppqz/2j9nxa9Ppi6sYUsUgSa2yy/90f8c2342c9578aafda73df2ffec24c1/thebiglebowskifeat-770x433.jpg?w=516&amp;h=290&amp;q=100 1.5x,
//images.ctfassets.net/9w4kzu6fppqz/2j9nxa9Ppi6sYUsUgSa2yy/90f8c2342c9578aafda73df2ffec24c1/thebiglebowskifeat-770x433.jpg?w=688&amp;h=387&amp;q=100 2x"><img alt="" width="344" height="193" src="//images.ctfassets.net/9w4kzu6fppqz/2j9nxa9Ppi6sYUsUgSa2yy/90f8c2342c9578aafda73df2ffec24c1/thebiglebowskifeat-770x433.jpg?w=344&amp;q=100" style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center center; opacity: 1; transition: opacity 0.5s ease 0s;"></picture>
```

**Comment by sidharthachatterjee** [comment]
Okay, just managed to reproduce this help from @pieh 

Turns out it filters the sizes out when 2x of the query width is _not_ less than actual width of the image 

It should do a `<=`, patching this in a min!

**Comment by sarahbethfederman** [comment]
Thanks for the quick response and PR!

**Comment by sidharthachatterjee** [comment]
You're welcome, @sarahbethfederman 

Happy to help! 🙂

**Comment by DSchau** [comment]
This has been closed in #11009.

If for some reason this isn't the case, please re-open and we'll work on a fix; thanks!
